File: frontend/ui/src/app/types/tutorials.ts
export type Tutorial = 'branding' | 'agents' | 'registry'

export interface TutorialStepData {

}

export interface TutorialProgressEvent {
  stepId: string;
  taskId: string;
  value?: any;
}

export interface TutorialProgressRequest extends TutorialProgressEvent {
  markCompleted?: boolean;
}

/*
{

}
 */

export interface TutorialTaskData {
  [key: string]: {
    value?: boolean | string | number
  };
}

export interface TutorialProgressData {
  [key: string]: TutorialTaskData;
}

export interface TutorialProgress {
  tutorial: Tutorial;
  // steps: TutorialStep[];
  createdAt?: string;
  completedAt?: string;
  // data?: TutorialProgressData;
  events: TutorialProgressEvent[];
}

File: internal/types/tutorials.go
package types

import "time"

type TutorialProgressEvent struct {
	StepID    string    `json:"stepId"`
	TaskID    string    `json:"taskId"`
	Value     any       `json:"value,omitempty"`
	CreatedAt time.Time `json:"createdAt"`
}

type TutorialProgressRequest struct {
	TutorialProgressEvent
	MarkCompleted bool `json:"markCompleted"`
}

type TutorialProgress struct {
	Tutorial    Tutorial                `db:"tutorial" json:"tutorial"`
	CreatedAt   time.Time               `db:"created_at" json:"createdAt"`
	Events      []TutorialProgressEvent `db:"events" json:"events,omitempty"`
	CompletedAt *time.Time              `db:"completed_at" json:"completedAt,omitempty"`
}
